### PR TITLE
Product sharing: Update UI for the AI sheet

### PR DESCRIPTION
**Pull request by itsmeichigo**
Part of #9867 

## Description

This PR updates the UI of the product sharing sheet with AI following the recent update [pe5sF9-1AH-p2#comment-2283]. Changes include:
- Updated the generate button style to `SecondaryLoadingButtonStyle` with a title for the loading state.
- Added a small button for showing the legal page.
- To better adapt with different text sizes, the two buttons are put together in an `AdaptiveStack`. The generate button is kept with a fixed size for non-accessibility font sizes in portrait mode. 
- Replaced `sizeCategory` with `dynamicTypeSize` as it's [deprecated](https://developer.apple.com/documentation/swiftui/environmentvalues/sizecategory) soon.
- Use `localizedMessage` for error to make it clearer and better match the behavior of product description generation.

## Testing instructions

- Log in to a public WPCom store with at least 1 published product.
- Tap to share the product, notice that the AI sheet matches the design in the aforementioned comment.
- Tap the (i) button, notice that a web view is presented showing the legal page for AI content.

## Screenshots

https://github.com/woocommerce/woocommerce-ios/assets/5533851/4d6b4fe9-c714-469d-9bd3-cfae1d0b5dd2

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/ViewRelated/ReusableViews/SwiftUI Components/ButtonStyles.swift [hunk 1]
```diff
@@ -31,10 +31,20 @@ struct SecondaryLoadingButtonStyle: PrimitiveButtonStyle {
...
+    /// Optional text when the button is in loading state
+    var loadingText: String?
+
+    /// Returns a `ProgressView` if the view is loading.
     ///
-    private var progressViewOverlay: ProgressView<EmptyView, EmptyView>? {
-        isLoading ? ProgressView() : nil
+    @ViewBuilder
+    private var progressViewOverlay: some View {
+        if isLoading {
```
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/ReusableViews/SwiftUI Components/ButtonStyles.swift]: super nit: how about `guard isLoading else { return nil }`?

**Review by jaclync — APPROVED** [review]
LGTM :shipit: 

I'm still seeing the generated message in quotes, but I think it's because the PR isn't based on the latest trunk.

File: WooCommerce/Classes/ViewRelated/ReusableViews/SwiftUI Components/ButtonStyles.swift (hunk 1 above)
Comment (itsmeichigo) [reply WooCommerce/Classes/ViewRelated/ReusableViews/SwiftUI Components/ButtonStyles.swift]: `guard` and `return nil` doesn't work with `@ViewBuilder`. I've replaced the `isLoading` check with `renderedIf` to make it look better in 1451fe1e80c85fb5c942bb531705e11ddd636a62.

**Comment by itsmeichigo** [comment]
> I'm still seeing the generated message in quotes, but I think it's because the PR isn't based on the latest trunk.

Correct! I'll merge trunk back to my branch next time before marking a PR as ready, sorry.

**Pull request merged**